postgres/lib/build.js

```diff
@@ -1,37 +1,12 @@
 const cds = require('@sap/cds')
 
-let BuildPlugin
-try {
-  ({ BuildPlugin } = require('@sap/cds-dk/lib/build'))
-} catch (e) {
-  if (e.code === 'ENOTFOUND') throw `No build plugin mechanism for @sap/cds-dk found. Please install @sap/cds-dk for development using 'npm i -D @sap/cds-dk@^7.3.0'`
-  else throw e
-}
-
-const { fs, path, rimraf } = cds.utils
-
-module.exports = class PostgresBuildPlugin extends BuildPlugin {
-  static hasTask() { // REVISIT: should be unnecessary -> plugin mechanism knows what to pull
```

Comment:
The `hasTask` impl has ensured that a postgres build task is only executed if a postgres database is used according to the effective cds configuration. E.g. a `sqlite` database might have been configured for the `development` profile. Of course, this might be an edge case as cds build will normally only be executed for the `production` and not for the `development` profile...

Reply:
Yes, unfortunately I also don't see another way to make it work properly w/ the profile technique where you switch between HANA/Postgres based on config profiles. I re-added the `hasTask` again.

Reply:
...unfortunately? - I think this is a perfect solution :)

Reply:
I think it's close to perfect, but ideally the plugin doesn't handle what the plugin orchestrator in `cap/cds` should do, deciding which plugins to pull.

I'd rather have opt-out mechanism like this instead, which is then also combinable with profiles:
```json
"cds": {
  "requires": {
      "plugins": {
        "[pg]":  {
           "@sap/cds-hana": false
        },
        "[hana]": {
          "@cap-js/postgres": false
        }
     }
  }
}
```

Opting out in non-centralized places like the build plugin introduces inconsistent behavior, e.g. the plugin will still show up as loaded with `DEBUG=plugins`.

Also, it reduces the number of functions you have to implement from 2 to 1, which is huge imo.

Reply:
But isn't this more a question of different lifecycles for `cds plugin` versus `cds build task`. Loading the `cds plugin` does not necessarily mean that a corresponding build task respectively its BuildPlugin counterpart will get executed.

I think the `cds add plugin` would behave similar. A `cds plugin` might get loaded, but the concrete `cds facet plugin` might not, as this is determined based on the concrete `cds add` command line parameters or some other facet dependencies.